packages/types/src/config.ts

```diff
@@ -724,6 +725,7 @@ export type ExperimentalSettings = {
     EnableRemoteClusterService: boolean;
     EnableAppBar: boolean;
     PatchPluginsReactDOM: boolean;
+    EnableChannelAutocomplete: boolean;
```

Comment:
Placed under experimental settings following a review from the API PR https://github.com/mattermost/mattermost-server/pull/22353#discussion_r1112078333